pkgs/development/python-modules/seahub/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv, lib, fetchFromGitHub, python3Packages, makeWrapper }:
+
+python3Packages.buildPythonPackage rec {
+  pname = "seahub";
+  version = "8.0.4";
+
+  src = fetchFromGitHub {
+    owner = "haiwen";
+    repo = "seahub";
+    rev = "v${version}-server";
+    sha256 = "0wi1m6k1km2n17s19jxg05jhmv7x59w92mijnay786ghhakd9y1r";
+  };
+
+  dontBuild = true;
+  doCheck = false; # disabled because it requires a ccnet environment
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  propagatedBuildInputs = with python3Packages; [
+    django
+    future
+    django-statici18n
+    django-webpack-loader
+    django-simple-captcha
+    django-picklefield
+    django-formtools
+    mysqlclient
+    dateutil
+    pillow
+    django_compressor
+    djangorestframework
+    openpyxl
+    requests
+    requests_oauthlib
+    pyjwt
+    pycryptodome
+    qrcode
+    pysearpc
+    seaserv
+  ];
+
+  installPhase = ''
+    cp -dr --no-preserve='ownership' . $out/
+    wrapProgram $out/manage.py \
```

Comment:
I don't see how to use `patchPythonScript` to bring `$out/thirdparts` into python path. Any idea?

Reply:
Disclaimer: I never got into a state where I could test the setup in practice.
But: Is there even a need for a `thirdparts` directory? That is just Seafile's way of vendoring certain libraries, which are all available directly from PyPi as well. So my approach was to package all of these python modules as normal python modules, and use `patchPythonScript` to make them being found at import time. No need to construct a separate `thirdparts` vendoring dir.

Reply:
This is not just vendoring, they rely on modified versions of the packages. See for instance this [PR](https://github.com/haiwen/seahub/pull/4191). Since there is no description, we don't known what they changed :shrug:

Reply:
Ugly but necessary then. I consider this to be resolved.